src/MCPClient/lib/clientScripts/email_fail_report.py

```diff
@@ -182,29 +180,38 @@ def store_report(content, unit_type, unit_name, unit_uuid):
     Report.objects.create(content=content, unittype=unit_type, unitname=unit_name, unitidentifier=unit_uuid)
 
 
-if __name__ == '__main__':
+def call(jobs):
     parser = argparse.ArgumentParser()
     parser.add_argument('-t', '--unitType', action='store', dest='unit_type', required=True)
     parser.add_argument('-i', '--unitIdentifier', action='store', dest='unit_uuid', required=True)
     parser.add_argument('-n', '--unitName', action='store', dest='unit_name', default='')
     parser.add_argument('-f', '--from', action='store', dest='from', default='ArchivematicaSystem@archivematica.org')
     parser.add_argument('--stdout', action='store_true', dest='stdout', default=False)
-    args = parser.parse_args()
-
-    to = get_emails_from_dashboard_users()
-    if not to:
-        logger.error('Nobody to send it to. Please add users with valid email addresses in the dashboard.')
-        sys.exit(1)
-    subject = 'Archivematica Fail Report for %s: %s-%s' % (args.unit_type, args.unit_name, args.unit_uuid)
-    efrom = 'ArchivematicaSystem@archivematica.org'
-
-    # Generate report in HTML and send it by email
-    content = get_content_for(args.unit_type, args.unit_name, args.unit_uuid, html=True)
-    send_email(subject, to, efrom, content)
-
-    if args.stdout:
-        print(content)
 
-    # Generate report in plain text and store it in the database
-    content = get_content_for(args.unit_type, args.unit_name, args.unit_uuid, html=False)
-    store_report(content, args.unit_type, args.unit_name, args.unit_uuid)
+    with transaction.atomic():
```

Comment:
Does this script need to have a db transaction?  It doesn't look like anything is written to the db here, so this line could be removed?

Reply:
Yeah, store_report is writing to the DB.  This might want the same treatment as the clamscan script--separate the bits that write to the database from the rest, so that we're not holding open the transaction while sending out emails.  We can look at this today as we'll be colocated :)

Reply:
I've addressed this one now.  It'll send all 100ish emails in a transaction after having sent out the emails.